src/app/ReadHandler.h

```diff
@@ -117,17 +124,25 @@ class ReadHandler : public Messaging::ExchangeDelegate
     // is larger than current self vended event number
     void MoveToNextScheduledDirtyPriority();
 
+    bool IsSubscription() { return mInteractionType == InteractionType::Subscribe; }
     bool IsInitialReport() { return mInitialReport; }
+    CHIP_ERROR OnSubscribeRequest(Messaging::ExchangeContext * apExchangeContext, System::PacketBufferHandle && aPayload);
+    void GetSubscriptionId(uint64_t & aSubscriptionId) { aSubscriptionId = mSubscriptionId; }
 
 private:
     enum class HandlerState
     {
-        Uninitialized = 0, ///< The handler has not been initialized
-        Initialized,       ///< The handler has been initialized and is ready
-        Reportable,        ///< The handler has received read request and is waiting for the data to send to be available
-        Reporting,         ///< The handler is reporting
+        Uninitialized = 0,     ///< The handler has not been initialized
+        Initialized,           ///< The handler has been initialized and is ready
+        Reportable,            ///< The handler has received read request and is waiting for the data to send to be available
+        Reporting,             ///< The handler is reporting
```

Comment:
Not clear whether this refers the act of creating a report, or already having sent a report. 

```suggestion
        AwaitingReportResponse,             ///< The handler has sent the report to the client and is awaiting a status response.
```

Reply:
Done, thanks